CI randomly fails many time in TestRepoStatsIndex

This failure occurs in various PRs randomly, including a PR which only changes one line document.

```
2022/03/19 18:52:07 ...xer/stats/indexer.go:39:populateRepoIndexer() [I] Populating the repo stats indexer with existing repositories
2022/03/19 18:52:07 ...xer/stats/indexer.go:85:populateRepoIndexer() [I] Done (re)populating the repo stats indexer with existing repositories
--- FAIL: TestRepoStatsIndex (5.66s)
    indexer_test.go:41: 
        	Error Trace:	indexer_test.go:41
        	Error:      	Not equal: 
        	            	expected: "65f1bf27bc3bf70f64657658635e66094edbcb4d"
        	            	actual  : ""
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual

        	            	@@ -1 +1 @@
        	            	-65f1bf27bc3bf70f64657658635e66094edbcb4d
        	            	+
        	Test:       	TestRepoStatsIndex
FAIL

coverage: 60.2% of statements

2022/03/19 18:52:13 ...age_stats_nogogit.go:86:func2() [E] Unable to open checker for 3aa73c3499bff049a352b4e265575373e964b89a. Error: failed to run attr-check. Error: exit status 128
	Stderr: fatal: Unable to read current working directory: No such file or directory

2022/03/19 18:52:13 ...it/repo_attribute.go:220:func1() [E] CheckPath returns error: context canceled
FAIL	code.gitea.io/gitea/modules/indexer/stats	12.712s
```

--- a/modules/indexer/stats/indexer_test.go
+++ b/modules/indexer/stats/indexer_test.go
@@ -5,12 +5,14 @@
 package stats
 
 import (
+	"context"
 	"path/filepath"
 	"testing"
 	"time"
 
 	repo_model "code.gitea.io/gitea/models/repo"
 	"code.gitea.io/gitea/models/unittest"
+	"code.gitea.io/gitea/modules/queue"
 	"code.gitea.io/gitea/modules/setting"
 
 	_ "code.gitea.io/gitea/models"
@@ -32,10 +34,14 @@ func TestRepoStatsIndex(t *testing.T) {
 	err := Init()
 	assert.NoError(t, err)
 
-	time.Sleep(5 * time.Second)
-
 	repo, err := repo_model.GetRepositoryByID(1)
 	assert.NoError(t, err)
+
+	err = UpdateRepoIndexer(repo)
+	assert.NoError(t, err)
+
+	queue.GetManager().FlushAll(context.Background(), 5*time.Second)
+
 	status, err := repo_model.GetIndexerStatus(repo, repo_model.RepoIndexerTypeStats)
 	assert.NoError(t, err)
 	assert.Equal(t, "65f1bf27bc3bf70f64657658635e66094edbcb4d", status.CommitSha)

